wdl/model/draft3/src/main/scala/wdl/model/draft3/elements/DeclarationElement.scala

```diff
@@ -27,3 +27,11 @@ object OutputDeclarationElement {
   * A declaration in an input block
   */
 final case class InputDeclarationElement(typeElement: TypeElement, name: String, expression: Option[ExpressionElement]) extends LanguageElement with WorkflowGraphNodeElement
+
+object DeclarationElement {
+  def unapply(languageElement: LanguageElement): Option[(TypeElement, String, Option[ExpressionElement])] = languageElement match {
```

Comment:
What is the intended use for this object? To me it seems to be used to conveniently access the members of one of the DeclarationElement types, but the LanguageElement type of the parameter is sort of confusing. I think a comment would be helpful.

Reply:
I'll definitely add a comment. This is a scala extractor (full description here: http://danielwestheide.com/blog/2012/11/21/the-neophytes-guide-to-scala-part-1-extractors.html)

TL;DR: 
* This is the function that allows the case match `case DeclarationElement(typeElement, name, Some(expr)) => ...` that's used [here](https://github.com/broadinstitute/cromwell/pull/3307/files#diff-3255c6301890ce6c0d4bb261ef95b340R21). 
* It's useful because it lets me treat all three of the types the same way in that match without having to retype the `=> ...` for each individual case class. 
* It takes a `LanguageElement` parameter because that's the type I'm matching against [here](https://github.com/broadinstitute/cromwell/pull/3307/files#diff-3255c6301890ce6c0d4bb261ef95b340R16).
* I made a note to myself to add a `case _ => None` to this unapply otherwise I'll get a MatchException thrown every time I try to run this against a `LanguageElement` which doesn't match one of these three cases.